src/main/java/org/dave/compactmachines3/gui/framework/GUI.java

```diff
@@ -100,15 +98,8 @@ public void drawTooltips(GuiScreen screen, int mouseX, int mouseY) {
         Widget hoveredWidget = getHoveredWidget(mouseX, mouseY);
         FontRenderer font = screen.mc.fontRenderer;
 
-        if(hoveredWidget != null && hoveredWidget.getTooltip() != null) {
-            if(hoveredWidget.getTooltip().size() > 0) {
-                GuiUtils.drawHoveringText(hoveredWidget.getTooltip(), mouseX, mouseY, width, height, 180, font);
-            }/* else {
-                List<String> tooltips = new ArrayList<>();
-                tooltips.add(hoveredWidget.toString());
-                GuiUtils.drawHoveringText(tooltips, mouseX, mouseY, width, height, 180, font);
-            }*/
+        if (hoveredWidget != null && hoveredWidget.hasTooltip()) {
+            GuiUtils.drawHoveringText(hoveredWidget.getTooltip(), mouseX, mouseY, width, height, -1, font);
```

Comment:
-1 here, instead of 180 in the previous iteration?

Reply:
Oops, didn't catch this when writing. I've refactored it to use the method builtin to `GuiScreen` that supports screen wrapping and switching sides depending on length, the same code that determines how item stack names are displayed.